mlflow/tracking/client.py

```diff
@@ -493,17 +493,21 @@ def update_model_version(self, name, version, description=None):
                                                                 description=description)
 
     @experimental
-    def transition_model_version_stage(self, name, version, stage):
+    def transition_model_version_stage(self, name, version, stage, archive_existing_versions=False):
```

Comment:
I think we should add some tests here: https://github.com/mlflow/mlflow/blob/master/tests/tracking/test_client.py

Reply:
I'll add tests.

Reply:
@andychow-db - i think we had a discussion (a really long time ago) about what to set the default for this API. one concern was that if we set it to be `False` it doesn't match the UI which is `True` by default. Do you remember what we decided there? i assume we want to set it to `False` here to not break any existing usages.

Reply:
I vaguely remember we said the UI would be true and the client would default to False to preserve behavior.

Reply:
ok